heartbeat/process/sender.go

```diff
@@ -131,12 +131,7 @@ func (s *Sender) SendHeartbeat() error {
 		trimLengths(hb)
 	}
 
-	hbBytes, err := s.marshalizer.Marshal(hb)
-	if err != nil {
-		return err
-	}
-
-	hb.Signature, err = s.singleSigner.Sign(s.privKey, hbBytes)
+	hb.Signature, err = s.singleSigner.Sign(s.privKey, hb.Pid)
```

Comment:
same thing here, might want to cache the signature

Reply:
Done.